key-generation/src/test/java/com/quorum/tessera/key/generation/AzureVaultKeyGeneratorTest.java

```diff
@@ -0,0 +1,127 @@
+package com.quorum.tessera.key.generation;
+
+import com.microsoft.azure.keyvault.models.SecretBundle;
+import com.quorum.tessera.config.ArgonOptions;
+import com.quorum.tessera.config.keypairs.AzureVaultKeyPair;
+import com.quorum.tessera.key.vault.KeyVaultService;
+import com.quorum.tessera.nacl.Key;
+import com.quorum.tessera.nacl.KeyPair;
+import com.quorum.tessera.nacl.NaclFacade;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.mockito.Mockito.*;
+
+public class AzureVaultKeyGeneratorTest {
+
+    private final String pubStr = "public";
+    private final String privStr = "private";
+    private final Key pub = new Key(pubStr.getBytes());
+    private final Key priv = new Key(privStr.getBytes());
+
+    private NaclFacade naclFacade;
+    private KeyVaultService keyVaultService;
+    private AzureVaultKeyGenerator azureVaultKeyGenerator;
+
+    @Before
+    public void setUp() {
+        this.naclFacade = mock(NaclFacade.class);
+        this.keyVaultService = mock(KeyVaultService.class);
+
+        final KeyPair keyPair = new KeyPair(pub, priv);
+
+        when(naclFacade.generateNewKeys()).thenReturn(keyPair);
+        when(keyVaultService.setSecret("id", "secret")).thenReturn(new SecretBundle());
+
+        azureVaultKeyGenerator = new AzureVaultKeyGenerator(naclFacade, keyVaultService);
+    }
+
+    @Test
+    public void keysSavedInVaultWithProvidedVaultId() {
+        final String vaultId = "vaultId";
+        final String pubVaultId = vaultId + "Pub";
+        final String privVaultId = vaultId + "Key";
+
+        final AzureVaultKeyPair result = azureVaultKeyGenerator.generate(vaultId, null);
+
+        verify(keyVaultService, times(2)).setSecret(any(String.class), any(String.class));
+        verify(keyVaultService, times(1)).setSecret(vaultId + "Pub", pub.toString());
```

Comment:
Please don't depend on the toString method on the Key objects. The fact that this has been used as a business method is an issue that I've addressed in another PR. As a general point toString should only ever be used to provide a readable representation of an instance not to return an encoded value or something functional.

Reply:
Now using the encodeToBase64 method